lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -135,37 +132,30 @@ TruncatedDistribution::TruncatedDistribution(const Distribution & distribution,
     const Interval & truncationInterval,
     const Scalar thresholdRealization)
   : DistributionImplementation()
+  , bounds_(truncationInterval)
 {
   setName("TruncatedDistribution");
-  if (distribution.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: can truncate only distribution with dimension=1, here dimension=" << distribution.getDimension();
   distribution_ = distribution;
   setThresholdRealization(thresholdRealization);
-  setDimension(1);
+  setDimension(distribution.getDimension());
```

Comment:
We should check the interval dimension against the distribution dimension

Reply:
ok, done